vms/platformvm/state/stakers.go

```diff
@@ -52,6 +66,12 @@ type CurrentStakers interface {
 	// Invariant: [staker] is not currently a CurrentDelegator
 	PutCurrentDelegator(staker *Staker)
 
+	// UpdateCurrentDelegator updates the [staker] describing a delegator to the
+	// staker set.
+	//
+	// Invariant: [staker] is a non deleted CurrentDelegator
+	UpdateCurrentDelegator(staker *Staker) error
```

Comment:
note: we don't need updating pending delegators (we plan to remove pending stakers completelly with continuous staking)